### PR TITLE
fix: update validation and conversion of timestamp to bingads offline conversion supported format

### PR DESCRIPTION
**Pull request by ItsSudip**
# Description

### Summary

This PR improves the handling of the `conversionTime` field for Bing Ads offline conversions and ensures robust test coverage for the new logic. The changes address issues with overly strict timestamp validation and align the system with Bing Ads' accepted formats, while also ensuring that the new logic is well-tested.

---

### Changes in `bulk_uploader.go`

- **Flexible Timestamp Parsing:**  
  The `conversionTime` field is now parsed using two formats:  
  1. **ISO 8601/RFC3339** (e.g., `2025-07-02T18:50:54Z`)  
  2. **Bing Ads UI format** (`mm/dd/yyyy hh:mm:ss AM/PM`, e.g., `7/2/2025 6:50:54 PM`)
- **Consistent Output:**  
  Regardless of the input format, the timestamp is always converted and output in Bing Ads' required format:  
  `mm/dd/yyyy hh:mm:ss AM/PM` (Go layout: `"2/1/2006 3:04:05 PM"`).
- **Clear Error Messaging:**  
  If the input does not match either accepted format, a clear error message is returned, specifying the valid formats.
- **Backward Compatibility:**  
  Existing integrations that send ISO 8601 timestamps will continue to work without modification.
- **No Dropped Valid Events:**  
  Valid events using either format are now accepted and processed, preventing unnecessary data loss.

---

### Changes in `bingads_test.go`

- **Test Coverage for Upload Logic:**  
  The test suite covers various upload scenarios, including:
  - Partial successes and failures in bulk uploads.
  - Handling of errors when obtaining upload URLs or uploading files.
  - Ensuring correct job ID tracking for successes and failures.
- **Test Coverage for Polling Logic:**  
  Tests verify correct handling of all Bing Ads bulk upload status responses, including:
  - Success, failure, partial failure, and in-progress states.
  - Correct aggregation of results when multiple request IDs are polled.
- **Test Coverage for Upload Stats:**  
  Tests ensure that upload statistics are correctly parsed and reported, including:
  - Aborted and succeeded job IDs.
  - Handling of error files and edge cases.
- **Test Environment Robustness:**  
  The test suite uses temporary directories and files to ensure isolation and repeatability, and cleans up resources after each test.

---

### Why

- **User Impact:**  
  Previously, only strict RFC3339 timestamps were accepted, causing valid Bing Ads events to be dropped if they used the more common Bing Ads UI format. This change ensures compatibility with Bing Ads documentation and user expectations.
- **Reliability:**  
  Improved test coverage ensures that the new logic is robust and that future changes will not break expected behavior.

---

### Success Criteria

- Both ISO 8601 and Bing Ads date-time formats are accepted as input for `conversionTime`.
- All outgoing events use the Bing Ads format for `conversionTime`.
- Invalid formats are rejected with a clear error message.
- No valid events are dropped due to timestamp formatting.
- All relevant upload and polling scenarios are covered by automated tests.

---

### Reference

- [Linear Issue INT-3891: Incorrect error message for Bing Ads offline conversions format](https://linear.app/rudderstack/issue/INT-3891/incorrect-error-message-for-bing-ads-offline-conversions-format)
- [Bing Ads Bulk Service Documentation](https://learn.microsoft.com/en-us/advertising/bulk-service/offline-conversion)

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/batchrouter/asyncdestinationmanager/bing-ads/offline-conversions/bingads_test.go [hunk 1]
```diff
@@ -523,7 +523,7 @@ var _ = Describe("Bing ads Offline Conversions", func() {
 			uploader := &BingAdsBulkUploader{
 				isHashRequired: true,
 			}
-			expectedResp := `{"message":{"fields":{"conversionCurrencyCode":"USD","conversionName":"Test-Integration","conversionTime":"2023-05-22T06:27:54Z","conversionValue":"100","email":"28a4da98f8812110001ab8ffacde3b38b4725a9e3570c39299fbf2d12c5aa70e","microsoftClickId":"click_id","phone":"8c229df83de8ab269e90918846e326c4008c86481393223d17a30ff5a407b08e"},"action":"insert"},"metadata":{"jobId":0}}`
+			expectedResp := `{"message":{"fields":{"conversionCurrencyCode":"USD","conversionName":"Test-Integration","conversionTime":"22/5/2023 6:27:54 AM","conversionValue":"100","email":"28a4da98f8812110001ab8ffacde3b38b4725a9e3570c39299fbf2d12c5aa70e","microsoftClickId":"click_id","phone":"8c229df83de8ab269e90918846e326c4008c86481393223d17a30ff5a407b08e"},"action":"insert"},"metadata":{"jobId":0}}`
```
Comment (koladilip) [inline router/batchrouter/asyncdestinationmanager/bing-ads/offline-conversions/bingads_test.go]: you said it should mm/dd/yyyy but example is dd/mm/yyyy

Comment (ItsSudip) [reply router/batchrouter/asyncdestinationmanager/bing-ads/offline-conversions/bingads_test.go]: Updated

File: router/batchrouter/asyncdestinationmanager/bing-ads/offline-conversions/bulk_uploader.go [hunk 2]
```diff
@@ -73,10 +73,20 @@ func (b *BingAdsBulkUploader) Transform(job *jobsdb.JobT) (string, error) {
...
 		return payload, fmt.Errorf("conversionTime field is not a string")
 	}
-	if _, err := time.Parse(time.RFC3339, conversionTimeStr); err != nil {
-		return payload, fmt.Errorf("conversionTime must be in ISO 8601 format like 2006-01-02T15:04:05Z07:00")
+
+	// Try RFC3339 first
+	parsedTime, parseErr := time.Parse(time.RFC3339, conversionTimeStr)
+	if parseErr != nil {
+		// Try Bing Ads format
+		parsedTime, parseErr = time.Parse("2/1/2006 3:04:05 PM", conversionTimeStr)
```
Comment (koladilip) [inline router/batchrouter/asyncdestinationmanager/bing-ads/offline-conversions/bulk_uploader.go]: can we give 2/13/2006?

Comment (ItsSudip) [reply router/batchrouter/asyncdestinationmanager/bing-ads/offline-conversions/bulk_uploader.go]: Yes

Comment (koladilip) [reply router/batchrouter/asyncdestinationmanager/bing-ads/offline-conversions/bulk_uploader.go]: Sorry I got confused, we should also use reference data Jan 2, 2006.
we want mm/dd/yyyy so we need to give 1/2/2006

**Review by koladilip — APPROVED** [review]
